libcloudforensics/gcp_test.py

```diff
@@ -16,28 +16,343 @@
+    self.assertEqual(disks[FAKE_DISK.name]['zone'], FAKE_DISK.zone)
+    self.assertEqual(disks[FAKE_BOOT_DISK.name]['zone'], FAKE_BOOT_DISK.zone)
+
+  @mock.patch('libcloudforensics.gcp.GoogleCloudProject.list_instances')
+  def test_get_instance(self, mock_list_instances):
+    """Test that an instance of a project can be found. """
+    mock_list_instances.return_value = {
+      FAKE_INSTANCE.name: {
+        'zone': FAKE_INSTANCE.zone
+      }
+    }
+    found_instance = FAKE_ANALYSIS_PROJECT.get_instance(FAKE_INSTANCE.name, FAKE_INSTANCE.zone)
+    self.assertIsInstance(found_instance, gcp.GoogleComputeInstance)
+    self.assertEqual(found_instance.project, FAKE_INSTANCE.project)
+    self.assertEqual(found_instance.name, FAKE_INSTANCE.name)
+    self.assertEqual(found_instance.zone, FAKE_INSTANCE.zone)
+    self.assertEqual(found_instance._data, FAKE_INSTANCE._data)
+    self.assertRaises(RuntimeError, FAKE_ANALYSIS_PROJECT.get_instance, 'non-existent-instance')
+
+  @mock.patch('libcloudforensics.gcp.GoogleCloudProject.list_disks')
+  def test_get_disk(self, mock_list_disk):
+    """Test that a disk of an instance can be found. """
+    mock_list_disk.return_value = {
+      FAKE_DISK.name: {
+        'zone': FAKE_DISK.zone
+      }
+    }
+    found_disk = FAKE_ANALYSIS_PROJECT.get_disk(FAKE_DISK.name)
+    self.assertIsInstance(found_disk, gcp.GoogleComputeDisk)
+    self.assertEqual(found_disk.project, FAKE_DISK.project)
+    self.assertEqual(found_disk.name, FAKE_DISK.name)
+    self.assertEqual(found_disk.zone, FAKE_DISK.zone)
+    self.assertRaises(RuntimeError, FAKE_ANALYSIS_PROJECT.get_disk, 'non-existent-disk')
+
+  @mock.patch('libcloudforensics.gcp.GoogleCloudProject.gce_operation')
+  @mock.patch('libcloudforensics.gcp.GoogleCloudProject.gce_api')
+  def test_create_disk_from_snapshot(self, mock_gce_api, mock_gce_operation):
+    """Test the creation of a disk from a snapshot. """
+    mock_gce_api.return_value.disks.return_value.insert.return_value.execute.return_value = None
+    mock_gce_operation.return_value = None
+
+    # create_disk_from_snapshot(snapshot)
+    disk_from_snapshot = FAKE_ANALYSIS_PROJECT.create_disk_from_snapshot(FAKE_SNAPSHOT)
+    self.assertIsInstance(disk_from_snapshot, gcp.GoogleComputeDisk)
+    self.assertEqual(disk_from_snapshot.name, self.__get_disk_name_for_snapshot(FAKE_SNAPSHOT))
+
+    # create_disk_from_snapshot(snapshot, disk_name='new-forensics-disk')
+    disk_from_snapshot = FAKE_ANALYSIS_PROJECT.create_disk_from_snapshot(FAKE_SNAPSHOT,
+                                                                         disk_name='new-forensics-disk')
+    self.assertIsInstance(disk_from_snapshot, gcp.GoogleComputeDisk)
+    self.assertEqual(disk_from_snapshot.name, self.__get_disk_name_for_snapshot(FAKE_SNAPSHOT,
+                                                                                 disk_name='new-forensics-disk'))
```

Comment:
Indentation looks off. May I suggest something along the lines of:

```
self.assertEqual(
  disk_from_snapshot.name,
  self.__get_disk_name_for_snapshot(FAKE_SNAPSHOT,disk_name='new-forensics-disk'))
```

Or again, if you know the value in advance, just use that string literal.

Reply:
Addressed. For this one I tried to improve readability. I also ran yapf on the code but I found it to break visibility in some places. Do we want to yapf everything?